Add variable for the ID of the account that owns the AMI

jsf9k opened the pull request:
## 🗣 Description

In this pull request I add an input variable for the ID of the account that owns the AMI.  The default value for this variable is `self`, so the module behaves just as it did previously unless you specify a different value for the new value.

## 💭 Motivation and Context

Previously the owner was hardcoded to `self`, which will not work when deploying to the COOL.  We need to specify the images account ID as the owner of the AMI in that case.

## 🧪 Testing

All the pre-commit hooks pass.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

jsf9k committed:
Add a variable for the ID of the account that owns the AMI_build

This variable is needed to deploy to the COOL.

jsf9k committed:
Use the new AMI owner account ID variable when looking up the AMI

felddy commented on ami.tf:
```diff
@@ -32,6 +25,8 @@ data "aws_ami" "openvpn" {
   }
 
-  owners      = ["self"]
+  owners = [
```
I didn't realize this is an array.  This raises interesting questions / solutions.

felddy reviewed (approved):
Yup.  This is important and should be back ported to the skeleton.

dav3r reviewed (approved):
Yup yup. 👍

mcdonnnj reviewed (approved):
👍

jsf9k committed:
Merge branch 'develop' into improvement/add-variable-for-ami-account

jsf9k committed:
Attempt fix for broken terraform-docs install

jsf9k committed:
Specify the go version

This is required to get around the fact that terraform-docs does not
install with the default version of go that is present.

jsf9k committed:
Go ahead and start using v2 of the setup-go action, even though it's only in beta right now

jsf9k committed:
Merge pull request #14 from cisagov/bugfix/fix-broken-terraform-docs-install

Fix broken terraform-docs install

jsf9k committed:
Merge remote-tracking branch 'skeleton-tf-module/develop' into improvement/add-variable-for-ami-account
